xbmc/cores/VideoPlayer/DVDSubtitles/SubtitleParserWebVTT.cpp

```diff
@@ -32,19 +32,36 @@ bool CSubtitleParserWebVTT::Open(CDVDStreamInfo& hints)
   if (!Initialize())
     return false;
 
+  CLog::LogF(LOGINFO, "Initialized WebVTT Parser");
+
   CWebVTTHandler m_webvttHandler;
 
   if (!m_webvttHandler.Initialize())
     return false;
 
-  // Get the first chars to check WebVTT signature
-  if (!m_webvttHandler.CheckSignature(m_pStream->Read(10)))
+  CLog::LogF(LOGINFO, "Initialized WebVTT Handler");
+
+  std::string line;
+
+  // WebVTT signature is according to spec length-variable and stops
+  // with two consecutive line terminators.
+  // Get the first line to check WebVTT signature.
+  if(!(m_pStream->ReadLine(line) && m_webvttHandler.CheckSignature(line)))
+  {
+    return false;
+  }
+
+  //Check if we have a second empty line as necessitated by spec.
+  //More empty lines might follow but will be handled by line decode.
+  if(!(m_pStream->ReadLine(line) && line.empty()))
```

Comment:
imo this should not invalidate the file,
here yes we follow webvtt specs, but, we are more relaxed about accepting formats that dont strictly follow the standards, usually this can happens with downloaded VOD streaming files and then if the second line isnt empty this can create a regression

Reply:
This one I added as a possibility but agree parsers should attempt to be as relaxed as feasible without major performance penalties and it is unnecessarily strict and will remove.